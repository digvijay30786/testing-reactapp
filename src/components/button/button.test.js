import { render, screen,fireEvent} from '@testing-library/react';
import Button from './button';
import Renderer from 'react-test-renderer';





describe("button.jsx testing", () => {

    test('button should we render without errors', () =>
    {
        const text = "ADD";
        render(<Button onClick={() => { }} label={text} />);
        
        const data = screen.getByTestId('btn');

        expect(data.textContent).toBe(text);

    });



    it("count the click when click the div", () => {
        const fn = jest.fn();
        const text = "CLICK ME";
        render(<Button onClick={fn} label={text} />);
        const btn = screen.getByTestId('btn');
        fireEvent.click(btn);

        expect(fn).toHaveBeenCalled();
        expect(fn).toHaveBeenCalledTimes(1);

        fireEvent.click(btn);
        fireEvent.click(btn);
        fireEvent.click(btn);

        expect(fn).toHaveBeenCalledTimes(4);
    });

    it("Take a snap Shot", () => {
        const snapshot = Renderer.create(
            <Button onClick={()=>{}} label="Button"/>
        );
        
        expect(snapshot.toJSON()).toMatchSnapshot();
    })
    
})
