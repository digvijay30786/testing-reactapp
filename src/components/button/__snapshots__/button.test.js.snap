// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button.jsx testing Take a snap Shot 1`] = `
<div
  data-testid="btn"
  onClick={[Function]}
>
  Button
</div>
`;
